lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -121,23 +121,24 @@ class window.Turbolinks
     for k,v of options.headers
       xhr.setRequestHeader k, v
 
-    xhr.onload = ->
-      if xhr.status >= 500
-        Turbolinks.fullPageNavigate(url)
-      else
-        Turbolinks.loadPage(url, xhr, options)
-      xhr = null
-
-    xhr.onerror = ->
-      # Workaround for sinon xhr.abort()
-      if xhr.statusText == "abort"
+    new Promise (resolve, reject) =>
+      xhr.onload = ->
+        if xhr.status >= 500
+          Turbolinks.fullPageNavigate(url)
+          reject()
```

Comment:
I agree that HTTP500+ should be `reject`, but what about 400+?  I think those are kinda fail-y too, but we just `resolve` them in this implementation.  Perhaps you want another code path:

```
if 500+
  fullNavigate & reject
if 400~499
  loadPage & reject
else
  loadPage & resolve
```
?

Not 100% sure but figured I'd add a comment about it